### PR TITLE
Support LOG_FORMAT env variable.

### PR DESCRIPTION
**Pull request by vincent-herlemont**
Related to https://github.com/nginx-proxy/nginx-proxy/discussions/2133

### CODE REVIEW

File: nginx.tmpl [hunk 1]
```diff
@@ -210,10 +211,7 @@ map $proxy_x_forwarded_proto $proxy_x_forwarded_ssl {
 
 gzip_types text/plain text/css application/javascript application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
 
-log_format vhost '$host $remote_addr - $remote_user [$time_local] '
-                 '"$request" $status $body_bytes_sent '
-                 '"$http_referer" "$http_user_agent" '
-                 '"$upstream_addr"';
+log_format vhost '{{ $globals.log_format }}';
```
Comment (rhansen) [inline nginx.tmpl:214]: Given there is only one user of the value I don't think we need to store it in `$globals`.

```suggestion
log_format vhost '{{ or $globals.Env.LOG_FORMAT "$host $remote_addr - $remote_user [$time_local] \"$request\" $status $body_bytes_sent \"$http_referer\" \"$http_user_agent\" \"$upstream_addr\"" }}';
```

File: test/test_log_format.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import pytest
+
+def test_log_format(docker_compose, nginxproxy):
+    r = nginxproxy.get("http://unknown.nginx-proxy.tld/port")
```
Comment (rhansen) [inline test/test_log_format.py:4]:
```suggestion
    r = nginxproxy.get("http://nginx-proxy.test/port")
```

File: test/test_log_format.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    assert r.status_code == 200
+    assert r.text == "answer from port 81\n"
+    sut_container = docker_compose.containers.get("test_log_format_sut_1")
```
Comment (rhansen) [inline test/test_log_format.py:7]: The container name should match the `container_name` value in the Docker compose file:

```suggestion
    sut_container = docker_compose.containers.get("sut")
```

File: test/test_log_format.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# GIVEN a webserver with VIRTUAL_HOST set to web1.tld
+web1:
+  image: web
+  expose:
+    - "81"
+  environment:
+    WEB_PORTS: 81
+    VIRTUAL_HOST: web1.tld
```
Comment (rhansen) [inline test/test_log_format.yml:8]: Client requests to names containing `nginx-proxy` will have their DNS lookups hijacked during testing to point to the nginx-proxy container:

```suggestion
    VIRTUAL_HOST: nginx-proxy.text
```

File: test/test_log_format.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+# WHEN nginx-proxy runs with DEFAULT_HOST set to web1.tld
+sut:
```
Comment (rhansen) [inline test/test_log_format.yml:11]: Specify the container name to guarantee you can successfully look it up from Python:

```suggestion
sut:
  container_name: sut
```

File: test/test_log_format.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  image: nginxproxy/nginx-proxy:test
+  volumes:
+    - /var/run/docker.sock:/tmp/docker.sock:ro
+  environment:
+    DEFAULT_HOST: web1.tld
```
Comment (rhansen) [inline test/test_log_format.yml:16]: No need to set `DEFAULT_HOST` if the hostname in the query matches.

```suggestion
```

**Review by rhansen — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Comment by vincent-herlemont** [comment]
@rhansen Thanks for the review, I applied your review.

File: test/test_log_format.yml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# GIVEN a webserver with VIRTUAL_HOST set to web1.tld
```
Comment (rhansen) [inline test/test_log_format.yml:1]:
```suggestion
```

File: test/test_log_format.yml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+web1:
+  image: web
+  expose:
+    - "81"
+  environment:
+    WEB_PORTS: 81
+    VIRTUAL_HOST: nginx-proxy.test
+
+# WHEN nginx-proxy runs with DEFAULT_HOST set to web1.tld
```
Comment (rhansen) [inline test/test_log_format.yml:10]:
```suggestion
```

**Review by rhansen — APPROVED** [review]

**Comment by vincent-herlemont** [comment]
@rhansen Thanks you! Did you know when is the next release with this feature is planned?

**Comment by rhansen** [comment]
That's up to @buchdag, although you can use the `latest` label which tracks `main`.

I forgot about updating the documentation. Would you mind opening a PR that mentions this new feature in `README.md`?

**Comment by buchdag** [comment]
@vincent-herlemont I'd like to have #1179 merged to complement #2146 before releasing a new version (and documentation for your PR's feature, as mentioned by @rhansen).

**Comment by vincent-herlemont** [comment]
@rhansen @buchdag Ok, follow up the documentation for this PR's feature https://github.com/nginx-proxy/nginx-proxy/pull/2152.

**Pull request merged**